#!/usr/bin/env python
# LandoltC configuration

PACKAGE = 'pandora_vision_landoltc'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Example : gen.add("int_param", int_t, 0, "An In.....", 50, 0, 100)
# name type level description def min max


#------------------------Threshold parameters------------------------
gen.add("gradientThreshold", double_t, 0,"A double Parameter", 60, 0, 100)
gen.add("centerThreshold", double_t, 0,"A double Parameter", 90, 0, 200)
gen.add("huMomentsPrec", double_t, 0,"A double Parameter", 0.3, 0, 1.0)
gen.add("adaptiveThresholdSubtractSize", int_t, 0, "An int Parameter", 2,1,10)
gen.add("visualization", bool_t, 0, "A boolean parameter for Visualization", False)

gen.add("timerThreshold", double_t, 0,"Threshold", 0.16, 0.0333, 0.2)

exit(gen.generate(PACKAGE, "landoltc_node", "landoltc_cfg"))
